experiment/queue_init.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Initializes the task queue, generates and pushes jobs into it for one
+experiment."""
+
+from redis import Redis
+from rq import Queue
+import sys
+import time
+
+from common import filesystem
+from experiment.task_module import build_task, run_task, measure_task
+
+
+def main():
+    """Run an experiment."""
+
+    # Initialize Redis server and task queues.
+    build_n_run_queue = Queue(name='build_n_run_queue', connection=Redis())
+    measure_queue = Queue(name='measure_queue', connection=Redis())
+
+    # Initialize the filestore.
```

Comment:
This filestore will store the experiment data, reports data, and, additionally, other source code for running experiment. When a worker would like to run a task, all required files for running a task (no matter that is build or run or measure) should be found at this place.

Reply:
The workers don't have to pull source code from other places if it starts from fuzzbench repo already.